diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/utils/AstNodeUtils.kt

```diff
@@ -163,6 +163,18 @@ fun ASTNode.findChildAfter(afterThisNodeType: IElementType, childNodeType: IElem
     return null
 }
 
+/**
+ * method that traverses previous nodes until it finds needed node or it finds stop node
+ * @return ASTNode?
+ */
+fun ASTNode.prevNodeUntilNode(stopNode: IElementType, checkNode: IElementType) : ASTNode? {
+    if (this.treePrev == null || this.treePrev.elementType == stopNode)
+        return null
+    if (this.treePrev.elementType == checkNode)
+        return this.treePrev
+    return this.treePrev.prevNodeUntilNode(stopNode, checkNode)
```

Comment:
You could use `ASTNode.siblings` from kotlin instead, which gives you a sequence instead of recursion